plugin/src/test/java/org/opensearch/ml/rest/MLCommonsRestTestCase.java

```diff
@@ -305,6 +305,16 @@ protected Response ingestModelData() throws IOException {
         return trainModelResponse;
     }
 
+    public void trainAsyncWithSample(Consumer<Map<String, Object>> consumer, boolean async) throws IOException {
+        String endpoint = "/_plugins/_ml/_train/sample_algo";
+        if (async) {
+            endpoint += "?async=true";
+        }
+        Response response = TestHelper
+            .makeRequest(client(), "POST", endpoint, ImmutableMap.of(), TestHelper.toHttpEntity(trainModelDataJson()), null);
+        verifyResponse(consumer, response);
```

Comment:
How about we sleep for few seconds to wait for the async task finish? I think 5 seconds should be enough.

Reply:
Good idea. We should be careful adding more and more ITs. I think we don't want to over test. The more IT we add, the more flaky it would be. I noticed some ITs do interfere each other.